### ISSUE TITLE
Cannot use Update-Dbainstance with -KB option

### ISSUE THREAD

**Issue by SajalBagchi** [sqlcollaborative/dbatools#6665]
Hi folks, I am trying to patch remote server using below command

Update-DbaInstance with -KB option as per the document.
It was working fine until last week but now it doesn't do anything. If I remove KB, it just works fine.
I tried from various machines with different target servers and still same behaviour.
Installer dbatool in a new machine, no luck from there too.

Current Dbatools version : 1.0.114 ( also tried old versions)

I am using KB because default parameters with no KB doesn't install security patch (GDR) ,it only supports SP and CU

Anyone else noticed this issue ?
Anyhelp is appreciated. Thank you.

**Comment by nvarscar** [comment]
The issue doesn't follow the template. Please re-open or submit all the information required as per https://github.com/sqlcollaborative/dbatools/issues/new?assignees=&labels=bugs_life&template=Bug_Report.md&title=%5BBug%5D

**Comment by wsmelton** [comment]
@nvarscar it is because it was migrated from another repo...so the issue template was not available when @SajalBagchi opened it.

**Comment by nvarscar** [comment]
Yes, that's what I figured. But still need more information 🤷 
Which is why I suggested re-opening or checking out the template. Specifically, I need error messages, if any. Also, @SajalBagchi any chance you can post the output of your command ran with a -Debug switch?

**Comment by SajalBagchi** [comment]
I am sorry new to to github so could not follow the template. Here are the details :



### Environmental information

PowerShell Version        : 5.1.14393.1944
dbatools latest installed : 1.0.113
Culture of OS             : en-US

#### SQL Server:

Microsoft SQL Server 2012 (SP4-OD) (KB4091266) - 11.0.7469.6 (X64) 
                Feb 28 2018 17:47:20 
                Copyright (c) Microsoft Corporation
                Enterprise Edition (64-bit) on Windows NT 6.3 <X64> (Build 9600: ) (Hypervisor)

us_english

### Report




#### Host used

- [X ] powershell.exe
- [X ] ISE
- [ ] VS Code
- [ ] Other (please specify)

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [X ] Still buggy in powershell.exe

#### Errors Received
VERBOSE: [17:15:42][Resolve-DbaNetworkName] Resolving <server_name> using .NET.Dns GetHostEntry
VERBOSE: [17:15:42][Resolve-DbaNetworkName] Resolving 10.XXX.XX.XXX using .NET.Dns GetHostByAddress
VERBOSE: [17:15:42][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [17:15:42][Get-DbaCmObject] [<server_name>] Retrieving Management Information
VERBOSE: [17:15:42][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM
VERBOSE: [17:15:43][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM - Success
VERBOSE: [17:15:46][Resolve-DbaNetworkName] Resolving server_name using .NET.Dns GetHostEntry
VERBOSE: [17:15:48][Resolve-DbaBuild] Looking for 11.0.7493
VERBOSE: [17:15:48][Resolve-DbaBuild] We have 91 builds in store for this Release
VERBOSE: [17:15:48][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [17:15:48][Get-DbaCmObject] [<server_name>] Retrieving Management Information
VERBOSE: [17:15:48][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM
VERBOSE: [17:15:49][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM - Success
VERBOSE: [17:15:50][Update-DbaInstance] Attempting to configure CredSSP for remote connections
VERBOSE: [17:15:50][Initialize-CredSSP] Configuring remote host to use CredSSP
VERBOSE: [17:15:53][Get-DbaCmObject] Configuration loaded | Cache disabled: False
VERBOSE: [17:15:53][Get-DbaCmObject] [<server_name>] Retrieving Management Information
VERBOSE: [17:15:53][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM
VERBOSE: [17:15:54][Get-DbaCmObject] [<server_name>] Accessing computer using Cim over WinRM - Success
VERBOSE: [17:15:54][Resolve-DbaBuild] Looking for KB 4532098
VERBOSE: [17:15:54][Resolve-DbaBuild] We have 91 builds in store for this Release
VERBOSE: [17:15:54][Get-SqlInstanceUpdate] Found applicable upgrade for SQL2012 to SP4 (KB4532098)
VERBOSE: [17:15:54][Get-SqlInstanceUpdate] Current SQL2012 version 11.0.7493 on computer [<server_name>] matches or already higher than target version 11.0.7493

$error[0] | select *` -- did not return anything

In above verbose, I have replaced Server Name with <Server_Name> and masked IP address
-->

#### Steps to Reproduce

Update SQL 2012 GDR update KB4532098  (but it could be any Target KB)
1. Execute Update-DbaInstance -ComputerName <Server_Name>-KB KB4532098 -Path <Setup Path> -Restart -Confirm:$false
2. Once Update is done, try running the same command again, ideally it should throw a warning like below from Get-SqlInstanceUpdate but that's not happening and it is terminating the script to move further.

WARNING: [14:32:44][Get-SqlInstanceUpdate] 11.0.7469 on computer [<server_name>] is already the latest available.

Another issue is, if we try running below command without KB as below, it does not detect security GDR KB4532098 as latest patch and gives above warning. (Reason I had to use KB parameter)

Update-DbaInstance -ComputerName Server1 -Path \\network\share -Restart -Confirm:$false

#### Expected Behavior



#### Actual Behavior

**Comment by nvarscar** [comment]
The output seems reasonable to me - it fails to install the KB that is already installed on the target machine: KB version and machine versions match - 11.0.7493.

> Current SQL2012 version 11.0.7493 on computer [<server_name>] matches or already higher than target version 11.0.7493

Also, to install a KB, it has to exist in the dbatools build reference file - that usually lags behind for a few days. If you're trying to install the KB that is not added yet, you'd get an error.

So I'm still confused what is the issue here. Could you please add Expected & Actual behaviour?

**Comment by SajalBagchi** [comment]
Expected Behaviour 1 : powershell is terminating and do not goto next scripts when installing with KB option. Also, on server with patch already installed, the warning message does not appear (if KB option is used). I could only see that information when Verbose/debug option is set.

Expected Behaviour 2 :  if we do not use KB option, it does not install security patch (GDR) but SP and CU (example 6 in dbatool doc)

Actual Behaviour 1 : PS should not terminate
Actual Behaviour 2:  Security/GDR patch should be considered if we do not specify KB

**Comment by SajalBagchi** [comment]
Build reference file is upto date and indexes have been updated regularly

**Comment by nvarscar** [comment]
Actual Behaviour 1 - if I understood correctly, you want the function to display a warning or throw an error when the patch is already installed. The way I see it, you're defining a level at which you want your instance, and it's not a reason to display a warning or stop execution if the instance already matches this criteria. In a scenario, where multiple servers are being patched, one server that already has the patch installed shouldn't prevent others from getting it. It would be possible to add a warning if the execution is a no-op for a particular server, but it would never be a terminating error. For your purposes you should just capture the command output and see if there were any actions done, something like this:
```powershell
$results = Update-DbaInstance ...
if ($results.Count -lt 1) { throw "No action performed" }
```

Actual Behaviour 2 - that is by design and won't be changed, since not everyone wants to install Security/GDR patches. If you want to install those - that's exactly why -KB parameter was implemented.
